docs/blog/posts/2024/07-26 asynchronous message support.md

```diff
@@ -0,0 +1,144 @@
+    for message in response.get("Messages", []):
+        if process_message(message):
+            sqs.delete_message(
+                QueueUrl=queue_url,
+                ReceiptHandle=message["ReceiptHandle"],
+            )
+```
+
+In this example, the `process_message` function processes messages from an SQS queue and calls the `delete_user` function to delete the user from the database and associated files. The `main` function listens for messages from the SQS queue and processes them using the `process_message` function.
+
+Here’s an example of a Pact test for this consumer:
+
+```python
+import json
+
+from pact.v3 import Pact
+
+from my_consumer import process_message
+
+def handler(body: str | bytes | None, metadata: dict[str, Any]) -> None:
+    message = json.loads(body)
+    process_message(message)
+
+pact = Pact(
+    consumer="deleteUserService",
+    provider="someProvider",
+).with_specification("V3")  # (1)
+
+(
+    pact
+    .upon_receiving("a request to delete a user", "Async")
+    .with_body(
+        json.dumps({
+            "action": "delete_user",
+            "user_id": "123",
+        })
+    )  # (2)
+)
+
+pact.verify(handler, "Async")
+```
+
+1.  Support for asynchronous messages starts in version 3 of the Pact specification.
+2.  No `will_respond_with` method exists for asynchronous messages since there’s no response expected.
+
+This example highlights how the verification of asynchronous messages differs from HTTP interaction. As the transport layer is abstracted away, a `handler` function is required to parse the raw message string or bytes, and pass it to the underlying function that processes the message.
+
+The `handler` would also typically be responsible for mocking the underlying systems that the consumer interacts with, such as the database or file system. This allows the consumer to be tested in isolation, without relying on external services. Furthermore, the mocked systems can then be inspected to verify that the consumer has performed the expected actions.
+
+## Provider Example
+
+For context of asynchronous messages, the provider is the service that sends the message and might be referred to as the _publisher_ or _producer_. Since the contract is defined by the consumer, the Pact provider test simply has to verify that the messages it sends meet the expectations of the consumer.
+
+As the underlying protocol is abstracted away, Pact uses a local HTTP server to receive the messages that the provider sends. The provider test for the above consumer might look something like this:
+
+```python
+from pact.v3 import Verifier
+
+class Provider:
+    """
+    A simple HTTP provider that sends messages to the consumer.
+
+    This would typically use the same underlying functions that would generate messages, except that instead of being sent into the message queue, they are sent to the consumer's HTTP server.
+    """
+
+provider = Provider()
+
+(
+    Verifier()
```

Comment:
Is it worth mentioning that this is a departure from the existing interfaces `MessageProvider`

Reply:
I personally want the blog post to just focus on what's coming, as opposed to how things were. 

I would like to create a more thorough upgrade guide once we are much closer to `v3` though.